src/librustc/diagnostics.rs

```diff
@@ -112,6 +112,40 @@ reference when using guards or refactor the entire expression, perhaps by
 putting the condition inside the body of the arm.
 "##,
 
+E0009: r##"
+In a pattern, all values that don't implement the Copy trait have to be binded the same way
+than the others. The goal here is to avoid to bind by-move and by-ref at the same time.
+Wrong example:
+struct X { x: (), }
+
+let x = Some((X { x: () }, X { x: () }));
+match x {
+    Some((_y, ref _z)) => {},
+    None => panic!()
+}
+
+You have two solutions:
+* implement the Copy trait on the X structure:
```

Comment:
Full sentences, maybe, since we can't format here. Something like:

You have two solutions:

Either you implement the Copy trait ...
(code)


or you bind the pattern's ....